src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Setting/SettingsManagerWrapper.cs

```diff
@@ -1,52 +1,62 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using NuGet.VisualStudio;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
     internal class SettingsManagerWrapper : ISettingsManager
     {
-        private readonly IVsSettingsManager _settingsManager;
+        private readonly AsyncLazy<IVsSettingsManager> _settingsManager;
 
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
         public SettingsManagerWrapper(IServiceProvider serviceProvider)
         {
-            _settingsManager = (IVsSettingsManager)serviceProvider.GetService(typeof(SVsSettingsManager));
-            Debug.Assert(_settingsManager != null);
+            _settingsManager = new AsyncLazy<IVsSettingsManager>(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                return (IVsSettingsManager)serviceProvider.GetService(typeof(SVsSettingsManager));
```

Comment:
>return [](start = 16, length = 6)

I also note that you're returning the result of this without checking whether it is null. Then later you write code that assumes that this lazy can never produce null (you'll throw NRE in that case). I suggest you assign the getservice(async) call's result to a local then call `Assumes.Present(local)` so that if it is null, you'll get a helpful exception message rather than a useless NRE.